test/features/resource_validation.feature

```diff
@@ -63,35 +63,18 @@ Feature: Resource validation
         When I log in
         And I open the new resource form for dataset "warandpeace"
         And I fill in "name" with "Test valid CSV create"
-        And I attach the file "valid.csv" to "upload"
+        And I attach the file "test.csv" to "upload"
+        And I attach the file "test_schema.json" to "schema_upload"
         And I fill in "description" with "Testing validation that should pass"
+        And I execute the script "document.getElementById('field-format').value='CSV'"
         And I press the element with xpath "//button[contains(@class, 'btn-primary')]"
         Then I should see "Test valid CSV"
-        When I wait for 5 seconds
         And I click the link with text that contains "Test valid CSV"
         Then I should see "Validation status"
         And I should see "success"
         And I should not see "failure"
         And I should see a validation timestamp
 
-
-    Scenario: As an editor, I can create a resource with an invalid CSV and see a failure status
```

Comment:
why delete?

Reply:
Resource_validation.feature runs sync validation, it's easier to test such behaviour via BDD tests. Async tested with unitests.

Reply:
Alex to restore BDD tests